dev: add note on being able to specify mavlink message telem rates in…

peterbarker opened the pull request:
… file

Hwurzburg commented on dev/source/docs/mavlink-requesting-data.rst:
```diff
@@ -146,6 +146,27 @@ Send a `COMMAND_LONG <https://mavlink.io/en/messages/common.html#COMMAND_LONG>`_
+Specifying Message Rates in a File
+----------------------------------
+
+At boot ArduPilot will populate the initial message intervals from files found in either ROMFS or in the filesystem (e.g. SD card).
```
at SD root? at APM dir root?

this is really cool....did not know it was in the code....is this master only or in stable? I assume it can be overwritten by the GCS....its a common problem on Dragonlink...I think we dismissed the option of stopping GCS from overwriting, hoping systems they would include it in DL docs, but still is a re-occurring problem for that system

also, if we could create an example for Dragonlink and include it, it would help a lot of people using that system

peterbarker replied:
@Hwurzburg this does not stop the GCSs from reconfiguring the telemetry rates using `SET_STREAM_RATE` or whatever.  It just permits absolute customisation of the stream rates for streaming messages, so you can prune out everything that doesn't matter to you.

GCSs should have options to turn off that auto-configuration of stream rates, as @khancyr mentions.  One extreme fix if a GCS doesn't behave is to turn on mavlink signing - can't reset streamrates if you don't know the key!

peterbarker replied:
Also, I haven't specified "APM directory" because that's not correct in the Linux HAL.  If you feel like adding more words, I'm happy to think about them :-)

Hwurzburg replied:
what dir in Linux systems?

peterbarker replied:
> what dir in Linux systems?

Varies based on board type, but typically the directory in which the binary is started.

Hwurzburg replied:
still needs specific instructions where to place the files for ROMFS or SD...by system if needed...otherwise will just cause support questions...

peterbarker replied:
I've added instructions on SD and ROMFS.

Hwurzburg commented on dev/source/docs/mavlink-requesting-data.rst:
```diff
@@ -146,6 +146,37 @@ Send a `COMMAND_LONG <https://mavlink.io/en/messages/common.html#COMMAND_LONG>`_
+
+.. code-block::
+
+   ROMFS message-intervals-chan0.txt message-intervals-chan0.txt
```
so, the first param is the file name that will be added into ROMFS and must be "message-intervals-chanX.txt" format for each channel, and the second param is the path to a text file with the date to be used in that file?   having the same name without some more explanation (assuming I have figured it out correctly) is confusing..
perhaps ROMFS message-intervals-chan0.txt <path to text file with data to load into the ROMFS file> ?

Hwurzburg replied:
also, need to add at very top as 5th method to set rates....but I can add after merge if you want

peterbarker commented:
On Thu, 30 Dec 2021, Henry Wurzburg wrote:

> so, the first param is the file name that will be added into ROMFS and must be "message-intervals-chanX.txt" format for each channel, and the second param is the path to a text file with
> the date to be used in that file? having the same name without some more explanation (assuming I have figured it out correctly) is confusing..
> perhaps ROMFS message-intervals-chan0.txt ?

That's correct on all counts.  I've added some path components and some 
better(?) text around this.  I've also moved it to below the explanation 
of the file names.

peterbarker commented:
On Thu, 30 Dec 2021, Henry Wurzburg wrote:

> also, need to add at very top as 5th method to set rates....but I can add after merge if you want

Added it.

Hwurzburg commented:
thanks...I hate the markup used here....a couple of expressions I wrote turned into just a ?

Hwurzburg commented:
okay has a couple of build errors...but easiest is to merge and I will fix immediately after....thanks
